ob2/util/job_limiter.py

```diff
@@ -0,0 +1,24 @@
+import logging
+
+from ob2.database import DbCursor
+from ob2.util.build_constants import QUEUED, IN_PROGRESS, FAILED
+from ob2.util.time import now_str
+
+MAX_JOBS_ALLOWED = 1
+
+def limit_queued_job(repo_name, build_name, limit):
+    try:
+        with DbCursor() as c:
+            c.execute('''SELECT count(*) FROM builds
+                         WHERE source = ? AND (status = ? OR status = ?)''',
+                      [repo_name, QUEUED, IN_PROGRESS])
+            count = int(c.fetchone()[0])
+            if count > limit:
+                message = "Cannot have more than {} builds in progress or queued.".format(limit)
+                c.execute('''UPDATE builds SET status = ?, updated = ?, log = ?
+                             WHERE build_name = ?''', [FAILED, now_str(), message, build_name])
+                return False
```

Comment:
It's a bit unintuitive that a function named 'limit_queued_job' would return a boolean.